File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int mxSm = INT_MIN;
        int n = nums.size();
        int currSm = 0;
        for(int i = 0; i<n; i++){
           currSm += nums[i];
           mxSm = max(currSm, mxSm);
           if(currSm < 0)
                currSm = 0;
        }
        return mxSm;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sm = Integer.MIN_VALUE;
        int n = nums.length;
        int currSm = 0;
        int startIndx = -1, endIndx = -1;
        int start = -1;
        List<Integer> list = new ArrayList<>();
        for(int i = 0; i<n ;i++){
            if(currSm == 0){
                start = i;
            }
            currSm += nums[i];

            if(currSm > sm){
                sm = currSm;
                startIndx = start;
                endIndx = i;
            }
            if(currSm < 0){
                currSm = 0;
            }
        }
        while(startIndx <= endIndx){
            list.add(nums[startIndx]);
            startIndx++;
        }
        System.out.println(list);
        return sm;
    }
}